ntoskrnl/ke/i386/mproc.c

```diff
@@ -0,0 +1,148 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Architecture specific source file to hold multiprocessor functions
+ * COPYRIGHT:   Copyright 2021 Justin Miller <justinmiller100@gmail.com>
+ *              Copyright 2021 Victor Perevertkin <victor.perevertkin@reactos.org>
+ */
+
+/* INCLUDES *****************************************************************/
+
+#include <ntoskrnl.h>
+// #define NDEBUG
+#include <debug.h>
+
+typedef struct _APINFO
+{
+    DECLSPEC_ALIGN(PAGE_SIZE) KGDTENTRY Gdt[128];
+    DECLSPEC_ALIGN(PAGE_SIZE) KIDTENTRY Idt[256];
+    DECLSPEC_ALIGN(16) UINT8 NMIStackData[DOUBLE_FAULT_STACK_SIZE];
+    KIPCR Pcr;
+    ETHREAD Thread;
+    KTSS Tss;
+    KTSS TssDoubleFault;
+    KTSS TssNMI;
+} APINFO, *PAPINFO;
+
+/* FUNCTIONS *****************************************************************/
+
+CODE_SEG("INIT")
+VOID
+NTAPI
+KeStartAllProcessors(VOID)
+{
+    PVOID KernelStack, DPCStack;
+    SIZE_T ProcessorCount = 0;
+    PAPINFO APInfo;
+
+    while (TRUE)
+    {
+        ProcessorCount++;
+        KernelStack = NULL;
+        DPCStack = NULL;
+
+        // Allocate structures for a new CPU.
+        APInfo = ExAllocatePoolZero(NonPagedPool, sizeof(APINFO), '  eK');
+        ASSERT(ALIGN_DOWN_POINTER_BY(APInfo, PAGE_SIZE) == APInfo);
+        if (!APInfo)
+        {
+            break;
+        }
+        KernelStack = MmCreateKernelStack(FALSE, 0);
+        if (!KernelStack)
+        {
+            break;
+        }
+        DPCStack = MmCreateKernelStack(FALSE, 0);
+        if (!DPCStack)
+        {
+            break;
+        }
```

Comment:
I would recommend to strip the curly braces in the 3 blocks above. 6 Locs wasted by that.